### PR TITLE
Add clangd config, format and review project issues

### PR DESCRIPTION
**Pull request by ryukinix**

### CODE REVIEW

File: examples/collision/square-platform.cpp [hunk 1]
```diff
@@ -1,39 +1,36 @@
...
-    {"quit", SDLK_ESCAPE},
-    {"up", SDLK_UP},
-    {"left", SDLK_LEFT},
-    {"right", SDLK_RIGHT},
-    {"down", SDLK_DOWN},
-    {"velocity_up", SDLK_w},
-    {"velocity_down", SDLK_s},
-    {"run", SDLK_LSHIFT}
-};
+std::map<const char*, SDL_Keycode> input_config = {
```
Comment (rafaelcn) [inline examples/collision/square-platform.cpp:30]: I only have concerns about this. I haven't seen how this displays on my local editor but on Github it looks weird.

**Review by rafaelcn — CHANGES_REQUESTED** [review]
I think there's room for improvement about the formatted.

File: examples/collision/square-platform.cpp (hunk 1 above)
Comment (ryukinix) [reply examples/collision/square-platform.cpp:30]: fixed @rafaelcn

File: examples/collision/square-platform.cpp [hunk 2]
```diff
@@ -289,30 +289,29 @@ void start() {
...
-        .w=2*block_size,
-        .h=block_size};
-    SDL_Rect block6 = {.x=SCREEN_WIDTH-2*block_size,
-        .y=SCREEN_HEIGHT-10*block_size,
-        .w=2*block_size,
-        .h=block_size};
+    SDL_Rect block1 = {
+        .x = 0, .y = SCREEN_HEIGHT - block_size, .w = 2 * block_size, .h = block_size
+    };
+    SDL_Rect block2 = { .x = SCREEN_WIDTH / 4,
```
Comment (ryukinix) [inline examples/collision/square-platform.cpp:295]: I think we can improve this as well, probably putting a comma in the end of the last struct field declaration.

File: examples/gfx/psx-doom-fire.cpp [hunk 3]
```diff
@@ -4,74 +4,51 @@
...
-    {183,175,47},
-    {183,183,47},
-    {183,183,55},
-    {207,207,111},
-    {223,223,159},
-    {239,239,199},
-    {255,255,255}
-};
+// int tiles[widthQuantity][heightQuantity];
+std::vector<std::array<int, 3>> colors = { { 0, 0, 0 }, { 31, 7, 7 },
```
Comment (ryukinix) [inline examples/gfx/psx-doom-fire.cpp:37]: here too

File: tests/TestCollision.cpp [hunk 4]
```diff
@@ -38,9 +37,9 @@ int main(void) {
     */
 
 
-    SDL_Rect r1 = {0, 0, 20, 20};
-    SDL_Rect r2 = {0, 20, 20, 20};
-    SDL_Rect r3 = {10, 30, 20, 20};
+    SDL_Rect r1 = { 0, 0, 20, 20 };
```
Comment (ryukinix) [inline tests/TestCollision.cpp:40]: I'm not sure if I want space after brackets for that kind of declaration.

**Comment by ryukinix** [comment]
Please review again @rafaelcn

File: examples/basics/square-moving.cpp [hunk 5]
```diff
@@ -16,13 +19,9 @@ using namespace std;
...
 // keys to InputHandler observe
 std::map<const char*, SDL_Keycode> input_config = {
-    {"quit", SDLK_ESCAPE},
-    {"up", SDLK_UP},
-    {"left", SDLK_LEFT},
-    {"right", SDLK_RIGHT},
-    {"down", SDLK_DOWN},
-    {"velocity_up", SDLK_w},
-    {"velocity_down", SDLK_s},
+    { "quit", SDLK_ESCAPE },     { "up", SDLK_UP },     { "left", SDLK_LEFT },
```
Comment (ryukinix) [inline examples/basics/square-moving.cpp:22]: I can't disable this if I enable this option: https://github.com/NewayPix/hazard-memory/pull/45/commits/fccfeb0ad7fbb2641a99b7cd9a1da35be411ef51

**Comment by ryukinix** [comment]
Any improvement will be made in an additional PR.

**Pull request merged**